### DISCUSSION TITLE
Why use RAM disk when linux caches to RAM anyway?

### DISCUSSION THREAD

**Discussion by pletharoe** [madMAx43v3r/chia-plotter#432] [Q&A]
Since Linux caches writes to RAM, you shouldn't need a RAM disk (assuming you're not doing any other IO).  Or am I missing something?

**Comment by SebMoore** [comment]
The biggest issue is that the plotter reads a huge amount of data too, because it has to sort data and compress it.  This means reads are a big problem, obviously, because the plotter doesn't constantly read the same file, so it can't be cached to ram. As such, the plotter has to keep going back to your ssd looking for files. Sure, maybe you could get a pseudo-ramdisk by increasing the time it takes to drop files out of the cache, but why not just use a regular ramdisk?

**Comment by madMAx43v3r** [comment]
If you have >384 GB RAM, then it works the same way yes. RAM disk just enforces the data which should be cached to be cached, reducing the amount of RAM needed to make it work.

**Reply by pletharoe** [reply]
Thanks, that's the answer I was looking for.

Now I just need to work out why it's only using 11 (ish) of my 128 threads.

**Comment by oleg-kn** [comment]
OS does not cache disk writes
